Lib/inspect.py

```diff
@@ -1790,9 +1790,6 @@ def _signature_get_partial(wrapped_sig, partial, extra_args=()):
             if param.kind is _POSITIONAL_OR_KEYWORD:
                 new_param = new_params[param_name].replace(kind=_KEYWORD_ONLY)
                 new_params[param_name] = new_param
-                new_params.move_to_end(param_name)
```

Comment:
This is a method specific to `OrderedDict` which moves the key to the end. Removal doesn't cause any test failure. Does this mean it was not tested before?

Reply:
+1 good catch, yep, it's specific to `OrderedDict`

Reply:
Yes, this was added by 3f73ca23cfe4e4058689bc5a46622c68ef1b6aa6 but I think it is not needed.

`param.kind` can be one of:
```
_POSITIONAL_ONLY 
_VAR_POSITIONAL 
_POSITIONAL_OR_KEYWORD
_KEYWORD_ONLY
_VAR_KEYWORD 
```

If it is `_POSITIONAL_ONLY` or `_POSITIONAL_OR_KEYWORD` it will get popped. 
If it is `_POSITIONAL_OR_KEYWORD`, `_KEYWORD_ONLY` or `_VAR_KEYWORD` it will be moved to the end of the OrderedDict.

As soon as `transform_to_kwonly` is set, it won't be removed and as `assert param.kind is not _POSITIONAL_ONLY` shows, there won't be any `_POSITIONAL_ONLY` anymore. This means that each remaining argument will either be popped or move to the end. Moving all arguments to the end won't change their relative order once the loop is finished so the `move_to_end` is actually not needed.

Maybe I just missed something, but 3f73ca23cfe4e4058689bc5a46622c68ef1b6aa6 added some tests and they are all still passing.